FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_Mutual_Auth/DemoTasks/MutualAuthHTTPExample.c

```diff
@@ -0,0 +1,461 @@
+     * maximum number of attempts or the maximum timeout value is reached. The
+     * function returns pdFAIL if the TCP connection cannot be established with
+     * the broker after configured number of attempts. */
+    xDemoStatus = connectToServerWithBackoffRetries( prvConnectToServer,
+                                                     &xNetworkContext );
+
+    if( xDemoStatus == pdPASS )
+    {
+        /* Set a flag indicating that a TLS connection exists. */
+        xIsConnectionEstablished = pdTRUE;
+
+        /* Define the transport interface. */
+        xTransportInterface.pNetworkContext = &xNetworkContext;
+        xTransportInterface.send = TLS_FreeRTOS_send;
+        xTransportInterface.recv = TLS_FreeRTOS_recv;
+    }
+    else
+    {
+        /* Log error to indicate connection failure after all
+         * reconnect attempts are over. */
+        LogError( ( "Failed to connect to HTTP server %.*s.",
+                    ( int32_t ) httpexampleAWS_IOT_ENDPOINT_LENGTH,
+                    democonfigAWS_IOT_ENDPOINT ) );
+    }
+
+    /*********************** Send HTTP request.************************/
+
+    if( xDemoStatus == pdPASS )
+    {
+        xDemoStatus = prvSendHttpRequest( &xTransportInterface,
+                                          HTTP_METHOD_POST,
+                                          httpexampleHTTP_METHOD_POST_LENGTH,
+                                          democonfigPOST_PATH,
+                                          httpexamplePOST_PATH_LENGTH );
+    }
+
+    /**************************** Disconnect. ******************************/
+
+    /* Close the network connection to clean up any system resources that the
+     * demo may have consumed. */
+    if( xIsConnectionEstablished == pdTRUE )
+    {
+        /* Close the network connection.  */
+        TLS_FreeRTOS_Disconnect( &xNetworkContext );
+    }
+
+    if( xDemoStatus == pdPASS )
+    {
+        LogInfo( ( "prvHTTPDemoTask() completed successfully. "
+                   "Total free heap is %u.\r\n",
+                   xPortGetFreeHeapSize() ) );
+        LogInfo( ( "Demo completed successfully.\r\n" ) );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvConnectToServer( NetworkContext_t * pxNetworkContext )
+{
+    BaseType_t xStatus = pdPASS;
+
+    TlsTransportStatus_t xNetworkStatus;
+    NetworkCredentials_t xNetworkCredentials = { 0 };
+
+    configASSERT( pxNetworkContext != NULL );
+
+    if( democonfigAWS_HTTP_PORT == 443 )
+    {
+        /* ALPN protocols must be a NULL-terminated list of strings. Therefore,
+         * the first entry will contain the actual ALPN protocol string while the
+         * second entry must remain NULL. */
+        char * pcAlpnProtocols[] = { NULL, NULL };
+        pcAlpnProtocols[ 0 ] = IOT_CORE_ALPN_PROTOCOL_NAME;
+
+        xNetworkCredentials.pAlpnProtos = pcAlpnProtocols;
```

Comment:
`pcAlpnProtocols` is defined within the scope of the `if` statement on line 310 and will go out of scope after line 319.  `xNetworkCredentials.pAlpnProtos` will be left pointing to that same location even though it may be overwritten.

Given that `pcAlpnProtocols[]` is an invariant, why not make it `const`?

```
static const char *pcAlpnProtocols[] = { IOT_CORE_ALPN_PROTOCOL_NAME, NULL };
```

Reply:
Thank you for catching that -- I've changed the variable to a `const` to address this.